tools/Customization/DevHome.Customization/ViewModels/DevDriveInsightsViewModel.cs

```diff
@@ -0,0 +1,605 @@
+    public async Task LoadAllDevDrivesInTheUI()
+    {
+        try
+        {
+            ExistingDevDrives = DevDriveManagerObj.GetAllDevDrivesThatExistOnSystem();
+            await UpdateListViewModelList();
+        }
+        catch (Exception /*ex*/)
+        {
+            // Log.Logger?.ReportError(Log.Component.SetupTarget, $"Error loading DevDriveViewModels data", ex);
+        }
+
+        ShouldShowShimmerBelowList = false;
+    }
+
+    /// <summary>
+    /// Loads all the DevDriveOptimizers and updates the UI with the results.
+    /// </summary>
+    public async Task LoadAllDevDriveOptimizersInTheUI()
+    {
+        try
+        {
+            if (!ExistingDevDrives.Any())
+            {
+                ExistingDevDrives = DevDriveManagerObj.GetAllDevDrivesThatExistOnSystem();
+            }
+
+            await UpdateOptimizerListViewModelList();
+        }
+        catch (Exception /*ex*/)
+        {
+            // Log.Logger?.ReportError(Log.Component.SetupTarget, $"Error loading DevDriveViewModels data", ex);
+        }
+
+        ShouldShowShimmerBelowOptimizerList = false;
+    }
+
+    /// <summary>
+    /// Loads all the DevDriveOptimizedCards and updates the UI with the results.
+    /// </summary>
+    public async Task LoadAllDevDriveOptimizedsInTheUI()
+    {
+        try
+        {
+            if (!ExistingDevDrives.Any())
+            {
+                ExistingDevDrives = DevDriveManagerObj.GetAllDevDrivesThatExistOnSystem();
+            }
+
+            await UpdateOptimizedListViewModelList();
+        }
+        catch (Exception /*ex*/)
+        {
+            // Log.Logger?.ReportError(Log.Component.SetupTarget, $"Error loading DevDriveViewModels data", ex);
+        }
+
+        ShouldShowShimmerBelowOptimizedList = false;
+    }
+
+    private void RemoveSelectedItemIfNotInUI(DevDrivesListViewModel listViewModel)
+    {
+        UpdateNextButtonState();
+    }
+
+    public async Task UpdateListViewModelList()
+    {
+        await _dispatcher.EnqueueAsync(async () =>
+        {
+            var curListViewModel = new DevDrivesListViewModel();
+            foreach (var existingDevDrive in ExistingDevDrives)
+            {
+                var card = await _devDriveViewModelFactory.CreateCardViewModel(DevDriveManagerObj, existingDevDrive);
+                curListViewModel.DevDriveCardCollection.Add(card);
+            }
+
+            AddListViewModelToList(curListViewModel);
+            DevDriveLoadingCompleted = true;
+            ShouldShowShimmerBelowList = true;
+        });
+    }
+
+    private string? GetExistingCacheLocation(string rootDirectory, string targetDirectoryName)
+    {
+        var fullDirectoryPath = rootDirectory + targetDirectoryName;
+        if (Directory.Exists(fullDirectoryPath))
+        {
+            return fullDirectoryPath;
+        }
+        else
+        {
+            var subDirPrefix = rootDirectory + "\\Packages\\PythonSoftwareFoundation.Python";

```

Comment:
How sure are you that this path will always be the path to look under?

Reply:
Worked with Mano for this. This is well documented in  [https://learn.microsoft.com/en-us/windows/dev-drive/#storing-package-cache-on-dev-drive](https://learn.microsoft.com/en-us/windows/dev-drive/#storing-package-cache-on-dev-drive).

Based on the source (store vs web) of the python that the user installs, we check the 2 locations. 
Again if python decides to change the prefix for the package name, we will have to rethink. But I think this would be rare.